sdk/iot/pnp/src/az_iot_pnp_client_telemetry.c

```diff
@@ -58,6 +58,7 @@ AZ_NODISCARD az_result az_iot_pnp_client_telemetry_get_publish_topic(
   AZ_PRECONDITION_VALID_SPAN(mqtt_topic, 0, false);
   AZ_PRECONDITION_IS_NULL(reserved);
   AZ_PRECONDITION_NOT_NULL(out_mqtt_topic);
+  (void)reserved;
```

Comment:
@CIPop  @danewalton 
Do we need to have all this methods with those unused parameters?  Why can't we just remove those parameters?
I have patched this for PRECONDITION=OFF, but I wonder if you can just have those removed if you don't use them

Reply:
I believe it is a placeholder for future fields. @jspaith added it for pnp necessities.